### PR TITLE
Replace FC with GEMM

### PR DESCRIPTION
**Pull request by duli2012**
Replace FC with GEMM

### CODE REVIEW

File: onnxmltools/convert/common/_topology.py [hunk 1]
```diff
@@ -605,7 +605,7 @@ def convert_topology(topology, model_name, doc_string, targeted_onnx):
     :return: a ONNX ModelProto
     '''
     if targeted_onnx != onnx.__version__:
-        raise RuntimeError(
+         raise RuntimeError(
```
Comment (wschin) [inline onnxmltools/convert/common/_topology.py]: Extra space?

File: onnxmltools/convert/coreml/operator_converters/neural_network/InnerProduct.py [hunk 2]
```diff
@@ -6,32 +6,31 @@
 
 from .....proto import onnx_proto
 from ....common._registration import register_converter
-
+from ....common._apply_operation import apply_reshape
```
Comment (wschin) [inline onnxmltools/convert/coreml/operator_converters/neural_network/InnerProduct.py]: Swap from `....common._registration import register_converterfrom` and `....common._apply_operation import apply_reshape`

File: onnxmltools/convert/coreml/operator_converters/neural_network/InnerProduct.py [hunk 3]
```diff
@@ -6,32 +6,31 @@
...
-        container.add_initializer(name_b, onnx_proto.TensorProto.FLOAT, shape_b, params.bias.floatValue)
+        container.add_initializer(name_c, onnx_proto.TensorProto.FLOAT, shape_c, params.bias.floatValue)
     else:
-        container.add_initializer(name_b, onnx_proto.TensorProto.FLOAT, shape_b, [0.] * shape_b[0])
-
-    attrs['axis'] = 1
-    attrs['axis_w'] = 1
-
-    container.add_node(op_type, inputs, outputs, **attrs)
+        container.add_initializer(name_b, onnx_proto.TensorProto.FLOAT, shape_c, [0.] * shape_b[0])
```
Comment (wschin) [inline onnxmltools/convert/coreml/operator_converters/neural_network/InnerProduct.py]: This initializer has been added in Line 21.

File: tests/end2end/test_single_operator_with_cntk_backend.py [hunk 4]
```diff
@@ -3,6 +3,7 @@
 # Licensed under the MIT License. See License.txt in the project root for
 # license information.
 # --------------------------------------------------------------------------
+import unittest
```
Comment (wschin) [inline tests/end2end/test_single_operator_with_cntk_backend.py]: Move `import unittest` right below `import numpy as np`

File: tests/end2end/test_single_operator_with_cntk_backend.py [hunk 5]
```diff
@@ -155,16 +156,21 @@ def _test_one_to_one_operator_core_channels_last(self, keras_model, x):
...
+        keras_model.compile(optimizer='adagrad', loss='mse')
 
-        self._test_one_to_one_operator_core(model, x)
+        coreml_model = coremltools.converters.keras.convert(keras_model)
+        onnx_model = onnxmltools.convert_coreml(coreml_model)
 
+        y_reference = keras_model.predict(x)
+        y_produced = _evaluate(onnx_model, x).reshape(N, D)
+
+        self.assertTrue(np.allclose(y_reference, y_produced))
```
Comment (wschin) [inline tests/end2end/test_single_operator_with_cntk_backend.py]: Add an empty line here.

File: tests/end2end/test_single_operator_with_cntk_backend.py [hunk 6]
```diff
@@ -384,9 +390,13 @@ def test_recursive_model(self):
         mapped1_2 = sub_model1(input1)
         mapped2_2 = sub_model2(input2)
         sub_sum = Add()([mapped1_2, mapped2_2])
-        model = Model(inputs=[input1, input2], outputs=sub_sum)
+        keras_model = Model(inputs=[input1, input2], output=sub_sum)
 
-        self._test_one_to_one_operator_core(model, [x, 2 * x])
+        coreml_model = coremltools.converters.keras.convert(keras_model)
```
Comment (wschin) [inline tests/end2end/test_single_operator_with_cntk_backend.py]: By calling _test_one_to_one_operator_core, we test conversion to from both of CoreML and Keras, this change means that we only test CoreML path now. I am not going to block you from checking in this important change but please do address my comments in another soon.

Comment (duli2012) [reply tests/end2end/test_single_operator_with_cntk_backend.py]: sure, I'll immediately produce another PR to incorporate all the comments.

File: onnxmltools/convert/coreml/operator_converters/neural_network/InnerProduct.py (hunk 3 above)
Comment (duli2012) [reply onnxmltools/convert/coreml/operator_converters/neural_network/InnerProduct.py]: Line 21 is an initializer for weights while line 28 is an initializer for bias, right?

Comment (wschin) [reply onnxmltools/convert/coreml/operator_converters/neural_network/InnerProduct.py]: Ok. Maybe name_b--->name_c?

Comment (duli2012) [reply onnxmltools/convert/coreml/operator_converters/neural_network/InnerProduct.py]: yes

File: tests/end2end/test_single_operator_with_cntk_backend.py (hunk 4 above)
Comment (duli2012) [reply tests/end2end/test_single_operator_with_cntk_backend.py]: Done

File: tests/end2end/test_single_operator_with_cntk_backend.py (hunk 6 above)
Comment (duli2012) [reply tests/end2end/test_single_operator_with_cntk_backend.py]: Will do it in another PR

**Pull request merged**